Updates to use kolibri-installer-gnome for an appified Kolibri experience

jamalex opened the pull request:
This removes most of the programs and patches that were bundled with
this package metadata. Instead, we use [kolibri-installer-gnome](https://github.com/learningequality/kolibri-installer-gnome), which provides most of the mechanisms we require.

bilelmoussaoui commented on org.learningequality.Kolibri.json:
```diff
@@ -1,11 +1,13 @@
-    "--persist=.kolibri"
+    "--persist=.kolibri",
+    "--socket=system-bus",
+    "--socket=session-bus"
```
Please run the app with `--log-session-bus` or `--log-system-bus` to get the list of the DBus you should add here.

jamalex replied:
Turns out neither was even necessary! I had assumed they would be needed for the D-Bus services I was adding (for search and activation), but it seem to be running fine without these permissions.

TingPing commented on src/org.learningequality.Kolibri.SearchProvider.service:
```diff
@@ -0,0 +1,3 @@
+[D-BUS Service]
+Name=org.learningequality.Kolibri.SearchProvider
+Exec=/usr/bin/flatpak run --command=/app/bin/kolibri_search_service.sh org.learningequality.Kolibri
```
This should be a normal `Exec` line, `flatpak` converts it into a `flatpak` command itself, I'm surprised this would ever work as-is.

jamalex replied:
Thanks for the note! In fact, you're right, it didn't yet work! :) I was uploading the in-progress work to the PR so I could point someone to it to get advice. New version that works coming shortly!

jamalex commented:
It's working now! I also updated the app icons and loading page to clean things up, and added content-type-specific icons for the search results:

![](https://files.slack.com/files-pri/T0KT5DC58-FQBE933FE/endless_fun_with_endless_search.gif)

Only issue I'm currently facing with this is that if Chrome is minimized, then the search result launching (which goes through `xdg-open`) will be loaded into the browser in the background, rather than focusing it, which isn't ideal.

indirectlylit commented:
FYI, I see that there are updated icons in the PR, but the screenshot posted still shows the old icon:

| new | old |
|--|--|
| ![image](https://user-images.githubusercontent.com/2367265/68532554-87141880-02ec-11ea-959a-570f5710ecd7.png) | ![image](https://user-images.githubusercontent.com/2367265/68532556-91cead80-02ec-11ea-947e-2588efd748f6.png)|

TingPing commented on patches/dbus_activation_service.py.patch:
```diff
@@ -0,0 +1,82 @@
+--- /dev/null	2019-11-07 08:53:04.471987251 -0800
++++ kolibri-0.12.8/kolibri/core/content/management/commands/dbus_activation_service.py	2019-11-08 16:34:21.992473978 -0800
+@@ -0,0 +1,79 @@
++import dbus
```
Assuming you wrote this, `python-dbus` is dead, don't use it. Use `Gio` via pygobject that provides DBus support:

- http://lazka.github.io/pgi-docs/#Gio-2.0/classes/DBusProxy.html#Gio.DBusProxy

TingPing commented on patches/dbus_activation_service.py.patch:
```diff
@@ -0,0 +1,82 @@
+--- /dev/null	2019-11-07 08:53:04.471987251 -0800
++++ kolibri-0.12.8/kolibri/core/content/management/commands/dbus_activation_service.py	2019-11-08 16:34:21.992473978 -0800
```
This file is going upstream in a future release?

TingPing commented on src/org.learningequality.Kolibri.SearchProvider.service:
```diff
@@ -0,0 +1,3 @@
+[D-BUS Service]
+Name=org.learningequality.Kolibri.SearchProvider
+Exec=/app/bin/kolibri_search_service.sh
```
```suggestion
Exec=kolibri_search_service.sh
```

TingPing replied:
If this is going upstream this needs to be a generated file with the correct path.

TingPing commented:
I am *extremely* confused on why this PR is here and not upstream. This needs code review not package review, none of this belongs here at all...

bilelmoussaoui commented:
> I am _extremely_ confused on why this PR is here and not upstream. This needs code review not package review, none of this belongs here at all...

I'm still wondering the same thing. This merge request isn't related to Flatpak/Flathub packaging at all.

jamalex commented:
(Sorry, I had merged another PR downstream that got included here, have backed that out so we can resolve this PR and then open that one as a new PR)

jamalex commented:
Thank you for the feedback. This is the first Flatpak package I've worked on, and I'm still getting familiar with the various conventions.

We're working to put things that were prototyped in the context of this flatpak package into the pipeline for the upstream package when they'd be more generally useful there and not specific to the Flatpak. Our general convention is to keep the primary upstream package platform-agnostic, and put OS-level or packaging-specific integrations into their respective packaging repositories.
(e.g. https://github.com/learningequality/kolibri-installer-android, https://github.com/learningequality/kolibri-installer-windows, https://github.com/learningequality/kolibri-installer-mac).

These specific commands are only intended to be used within the context of the Flatpak, being called by the Flatpak launcher scripts. The reason they're coded as management commands being patched into the package is that was the simplest way to have them run within the Django environment so that they could (in the case of the search provider) easily query the database via the ORM. If preferable, they could be moved to separate Python scripts that separately bootstrap the Django environment.

Or, if you'd prefer that this repository be kept purely to the manifest and basic supports around that, we could create a new upstream project that adds this Flatpak- and GNOME-specific functionality that can then be imported into the package.

TingPing commented:
This repo exists purely to build the software, this is not where development happens.

LeandroStanger commented on org.learningequality.Kolibri.json:
```diff
@@ -1,14 +1,52 @@
-  "runtime-version": "19.08",
-  "sdk": "org.freedesktop.Sdk",
+  "runtime": "org.gnome.Platform",
+  "runtime-version": "3.34",
```
Update runtime to 3.36

jamalex commented:
Updated this PR with a batch of fantastic work done by @dylanmccall to factor out the non-flatpak-specific elements here into upstream repositories, and add additional flatpak-specific code (for integrating with flatpak content extensions).

dylanmccall commented:
:+1: The test build works for me, including an upgrade from the existing stable version on flathub.
